### PR TITLE
fix(sdk): add avaxc family

### PR DESCRIPTION
**Pull request by fmartinAltamirano**
Added avaxc family to avoid calling the sweep api and allow sending the total funds in the wallet.

https://bitgoinc.atlassian.net/browse/STLX-7706

### CODE REVIEW

File: modules/core/src/v2/wallet.ts [hunk 1]
```diff
@@ -1055,7 +1055,7 @@ export class Wallet {
       params = params || {};
       common.validateParams(params, ['address'], ['walletPassphrase', 'xprv', 'otp'], callback);
```
Comment (CapnMigraine) [inline modules/core/src/v2/wallet.ts:1057]: nit: Add comment explaining this code path
```suggestion
// The sweep API endpoint is only available to utxo-based coins
```

File: modules/core/src/v2/wallet.ts [hunk 2]
```diff
@@ -1055,7 +1055,7 @@ export class Wallet {
...
 
-      if (['eth', 'xrp'].includes(self.baseCoin.getFamily())) {
+      if (['eth', 'xrp', 'avaxc'].includes(self.baseCoin.getFamily())) {
```
Comment (CapnMigraine) [inline modules/core/src/v2/wallet.ts:1058]: This should be the path for all non-utxo coins
```suggestion
      if (!self.baseCoin instanceof AbstractUtxoCoin) {
        // This path is for account-based coins
```

File: modules/core/src/v2/wallet.ts [hunk 3]
```diff
@@ -1055,7 +1055,9 @@ export class Wallet {
       params = params || {};
       common.validateParams(params, ['address'], ['walletPassphrase', 'xprv', 'otp'], callback);
 
-      if (['eth', 'xrp'].includes(self.baseCoin.getFamily())) {
+      // The sweep API endpoint is only available to utxo-based coins
+
+      if (!(self.baseCoin instanceof AbstractUtxoCoin)) {
```
[the remark this answers is not in the document]
Comment (fmartinAltamirano) [reply modules/core/src/v2/wallet.ts]: the change was made :+1:

[the remark this answers is not in the document]
Comment (tylerlevine) [reply modules/core/src/v2/wallet.ts]: > yarn install fails if we don't cast the left-hand side of the instanceof expression

worked fine for me, what was the issue here exactly?

**Pull request merged**